crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -296,6 +297,48 @@ where
             status_code: if receipt.success { Some(U64::from(1)) } else { Some(U64::from(0)) },
         };
 
+        // get all receipts for the block
+        let all_receipts = match self.client().receipts_by_block((meta.block_number).into())? {
+            Some(recpts) => recpts,
+            None => return Err(EthApiError::UnknownBlockNumber),
+        };
+
+        // get the previous transaction cumulative gas used
+        let prev_tx_cumulative_gas_used = match all_receipts.get((meta.index - 1) as usize) {
+            Some(prev_receipt) => prev_receipt.cumulative_gas_used,
+            // if meta.index == 0, then there is no previous transaction
+            None => 0,
+        };
+
+        // gas_used = CumulativeGasUsed(tx_index) - CumulativeGasUsed(tx_index - 1)
+        res_receipt.gas_used = Some(
+            U256::from(receipt.cumulative_gas_used)
+                .checked_sub(U256::from(prev_tx_cumulative_gas_used))
+                .ok_or(EthApiError::Unsupported("Overflow occured"))?,
+        );
+
+        let mut log_index: u32 = 0;
+        for (tx_idx, receipt) in all_receipts.into_iter().enumerate() {
+            for (tx_log_idx, log) in receipt.logs.into_iter().enumerate() {
+                if tx_idx == meta.index as usize {
```

Comment:
this is not quite accurate yet because this loop counts _all_ logs, even the logs after the transaction

we can optimize this a bit:

```rust
let mut receipts_iter = all_receipts.into_iter();

// take `index` items from iter (these are prev tx) and count num logs
// `all_receipts.next()` is no the receipt of the target transaction
 
```